src/adaptive/index.es

```diff
@@ -1,35 +1,20 @@
-import React, { PureComponent } from 'react';
+import React, { useEffect, useState } from 'react';
 import eventproxy from 'nebenan-helpers/lib/eventproxy';
 import { getMedia, media } from 'nebenan-helpers/lib/dom';
 import { getDisplayName, getForwardedComponent } from '../utils';
 
 
-const makeAdaptive = (Component, options = {}) => {
-  class AdaptiveComponent extends PureComponent {
-    constructor(props) {
-      super(props);
-      this.state = { isMobile: null };
-      this.detectDevice = this.detectDevice.bind(this);
-    }
-
-    componentDidMount() {
-      this.detectDevice();
-      this.stopListeningToResize = eventproxy('resize', this.detectDevice);
-    }
-
-    componentWillUnmount() {
-      this.stopListeningToResize();
-    }
+export const useIsMobile = () => {
+  const [isMobile, setIsMobile] = useState(false);
```

Comment:
not sure if `null` is needed, perhaps default to `null` to signify unset state?

Reply:
it looks like we needed `null` only for ssr

Reply:
if it's still in use please default to `null`